staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -341,6 +343,10 @@ func (rules *ClientConfigLoadingRules) IsDefaultConfig(config *restclient.Config
 	if err != nil {
 		return false
 	}
+
+	defaultConfig.Timeout = config.Timeout
+	defaultConfig.Impersonate = config.Impersonate
```

Comment:
a config that specifies impersonation options is definitely *not* a default config

Reply:
I dropped func IsDefaultConfig, and introuduce a new private function to check whether we should use in-cluster config.